apps/guide/settings/base.py

```diff
@@ -184,4 +238,228 @@
 WAGTAILIMAGES_IMAGE_MODEL = "custom_media.CustomImage"
 WAGTAILDOCS_DOCUMENT_MODEL = "custom_media.CustomDocument"
 
+pixel_limit = env.get("WAGTAILIMAGES_MAX_IMAGE_PIXELS")
+WAGTAILIMAGES_MAX_IMAGE_PIXELS = int(pixel_limit) if pixel_limit else 10_000_000
+
 DEFAULT_AUTO_FIELD = "django.db.models.BigAutoField"
+
+# AWS S3 buckets configuration
+# This is media files storage backend configuration. S3 is our preferred file
+# storage solution.
+# To enable this storage backend we use django-storages package...
+# https://django-storages.readthedocs.io/en/latest/backends/amazon-S3.html
+# ...that uses AWS' boto3 library.
+# https://boto3.amazonaws.com/v1/documentation/api/latest/index.html
+#
+# Three required environment variables are:
+#  * AWS_STORAGE_BUCKET_NAME
+#  * AWS_ACCESS_KEY_ID
+#  * AWS_SECRET_ACCESS_KEY
+# The last two are picked up by boto3:
+# https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#environment-variables
+if "AWS_STORAGE_BUCKET_NAME" in env:
+    # Add django-storages to the installed apps
+    INSTALLED_APPS = INSTALLED_APPS + ["storages", "wagtail_storages"]
+
+    # https://docs.djangoproject.com/en/stable/ref/settings/#default-file-storage
+    DEFAULT_FILE_STORAGE = "storages.backends.s3boto3.S3Boto3Storage"
+
+    AWS_STORAGE_BUCKET_NAME = env["AWS_STORAGE_BUCKET_NAME"]
+
+    # Disables signing of the S3 objects' URLs. When set to True it
+    # will append authorization querystring to each URL.
+    AWS_QUERYSTRING_AUTH = False
+
+    # Do not allow overriding files on S3 as per Wagtail docs recommendation:
+    # https://docs.wagtail.io/en/stable/advanced_topics/deploying.html#cloud-storage
+    # Not having this setting may have consequences in losing files.
+    AWS_S3_FILE_OVERWRITE = False
+
+    # Default ACL for new files should be "private" - not accessible to the
+    # public. Images should be made available to public via the bucket policy,
+    # where the documents should use wagtail-storages.
+    AWS_DEFAULT_ACL = "private"
```

Comment:
note: without wagtail_storages, this doesn't need to be private

Reply:
It still probably should be, so we rely on the bucket policy allocating permissions, not something in the files themselves.

Reply:
works for me